portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,20 +75,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     /** Returns comment limit value from the request, or -1 if user selected 'all' */
     private int getCommentLimit(HttpServletRequest request) {
         String limit_string = request.getParameter("comment-limit");
-        int comment_limit = 0;
-
-        try {
-            comment_limit = Integer.parseInt(limit_string);
-        } catch (NumberFormatException n) {
-            if (limit_string.equals("all")) {
-                comment_limit = -1;
-            }
-            else {
-                // Printing error messages?
-                System.out.println("Unexpected value '" + comment_limit + "' for comment limit.");
-            }
-        }
-
+        int comment_limit = Integer.parseInt(limit_string);
```

Comment:
Still a good idea to catch NumberFormatException here, just so you don't crash if something goes wrong with the inputs. Even though the exception is not required by the compiler to be caught.

If you have a way of seeing the output from the servlet, then Logging to System.err.println() is probably good enough. Note the "err" instead of "out".